code-challenges.test.js

```diff
@@ -1,69 +1,155 @@
 
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
+// const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+// // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+// const randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
+// // Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
 
+// describe("wordsCap", () => {
 
-// b) Create the function that makes the test pass.
+//     it("takes in an array of words and returns an array with all the words capitalized", () => {
+
+//         expect(wordsCap(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+//         expect(wordsCap(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango", "Deduction"])
+//     })
+// })
+
+// // good fail - ReferenceError: wordsCap is not defined
 
+      
+
+
+// // b) Create the function that makes the test pass.
+
+// // PC:
+// // create fn called wordsCap
+//     // takes array of words as input
+// //  .map() over array
+//     // capitalize first letter
+//         // find index 0 of each word
+//         // .toUppercase()
+//         // .substring() starting at [1]
+//         // glue them back together
+
+// const wordsCap = (array) => {
+//     array.map(first => {
+//         return first[0].toUpperCase() + first.substring(1)
+//     })
+// }
 
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
-// a) Create a test with expect statements for each of the variables provided.
+// a) Create a test with expect statements for each of the variables provided. 
 
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
+// const vowelTester1 = "learn"
+// // Expected output: 1
+// const vowelTester2 = "academy"
+// // Expected output: 0
+// const vowelTester3 = "challenges"
+// // Expected output: 2
 
+// describe("firstIndex", () => {
 
-// b) Create the function that makes the test pass.
+//     it("takes in a string and logs the index of the first vowel", () => {
+
+//       expect(vowelTester1(firstIndex)).toEqual("1")
+//       expect(vowelTester2(firstIndex)).toEqual("0")
+//       expect(vowelTester3(firstIndex)).toEqual("2")
+
+//     })
+//   })
+
+// // Passssssed!
+
+// // b) Create the function that makes the test pass.
+
+// // PS:
+// // create firstIndex fn
+// // takes a string (input)
+// // finds index 0 of given string
+//   // determines if vowel or not
+//   // console.logs it
+
+// const firstIndex = (string) => {
```

Comment:
This function works if we wanted the 0th index of each string (or first letter of each word). This question wants the index of the first vowel in each string.